### ISSUE TITLE
Odd sorting behavior

### ISSUE THREAD

**Issue by v36u** [theron-wang/VS2022-Editor-Support-for-Tailwind-CSS#67]
I've created a gradient with tailwind classes like this:
```
bg-gradient-to-r from-primary to-dark-blue
```
Full class list looks like this:
```
min-w-[258px] min-h-[345px] rounded-[1.2rem] transform-style-3d bg-gradient-to-r from-primary to-dark-blue h-full text-left shadow-xl transition-all duration-700 ease-in-out hover:cursor-help group-hover:rotate-y-180
```
However, when I save (and the sorting triggers, it becomes like this):
```
min-w-[258px] min-h-[345px] rounded-[1.2rem] transform-style-3d from-primary to-dark-blue h-full bg-gradient-to-r text-left shadow-xl transition-all duration-700 ease-in-out hover:cursor-help group-hover:rotate-y-180
```
What seems wrong:
- `h-full` was inserted in the gradient "section"
- `bg-gradient-to-r` was moved after `from-primary to-dark-blue`, which makes it harder to read fluently

**Comment by theron-wang** [comment]
Yep, this was definitely a strange bug, but these should be fixed in 1.5.3. Thanks!

**Comment by v36u** [comment]
Hi!

It sorts like this now which still seems odd:
```
transform-style-3d from-primary to-dark-blue group-hover:rotate-y-180 h-full min-h-[345px] min-w-[258px] rounded-[1.2rem] bg-gradient-to-r text-left shadow-xl transition-all duration-700 ease-in-out hover:cursor-help
```

By odd, I want to say that `bg-gradient-to-r` appears way later than `from-primary to-dark-blue`, but it would be expected to appear right before them (`bg-gradient-to-r from-primary to-dark-blue`)

**Comment by theron-wang** [comment]
I think this issue may have something to do with the extension treating custom colors incorrectly (i.e. it should work with `from-white` and `to-white`, for example). I’ll take a look into this when I get the chance.

**Comment by v36u** [comment]
Yes, I confirm that it works with built-in colors:
```
transform-style-3d group-hover:rotate-y-180 h-full min-h-[345px] min-w-[258px] rounded-[1.2rem] bg-gradient-to-r from-white to-white text-left shadow-xl transition-all duration-700 ease-in-out hover:cursor-help
```

**Comment by theron-wang** [comment]
I tried reproducing this issue with two cases:
```js
theme: {
  extend: {
    colors: {
      'primary': '#aaa',
      'dark-blue': '#aaa',
    }
  }
}
```
```js
theme: {
  colors: {
    'primary': '#aaa',
    'dark-blue': '#aaa',
  }
}
```

Both of these appear to sort correctly: `bg-gradient-to-r from-primary to-dark-blue` (note that `bg-gradient-to-r` is before everything). If possible, could you send your configuration file? Thanks.

**Comment by v36u** [comment]
Here it is:
<details>
<summary>`tailwind.config.js`</summary>

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./Components/**/*.{razor,cshtml,html}",
    "./Properties/**/*.resx"
  ],
  theme: {
    screens: {
      'sm': '576px',
      'md': '768px',
      'lg': '992px',
      'xl': '1200px',
      '2xl': '1400px',
    },
    fontSize: {
      'xs': '0.75rem',
      'sm': '0.875rem',
      'md': '1rem',
      'lg': '1.125rem',
      'xl': '1.25rem',
      '2xl': '1.5rem',
      '3xl': '2rem',
      '4xl': '2.75rem',
    },
    extend: {
      colors: {
        'primary': '#6e25d0',
        'secondary': '#992bc2',
        'smoke-white': '#eceff9',
        'dark-blue': '#221366',
        'green': '#36a22d',
        'gray': '#595e66',
        'lighter-gray': '#faf9fa',
        'darker-gray': '#7b7fa6',
        'dark-purple': '#1e0145',
      },
      fontSize: {
        '2xs': '0.6rem',
      },
      height: {
        '30p': '30%',
      },
      boxShadow: {
        'z-depth-5': '0px 5px 10px rgb(153 43 194 / 30%)',
        'z-depth-5-2': '0px 5px 10px rgb(153 43 194 / 50%)',
        'z-depth-4': '0 5px 11px 0 rgb(0 0 0 / 18%)',
      },
      borderWidth: {
        '1': '1px',
      },
      spacing: {
        '0.8': '0.8rem',
      },
      margin: {
        'negative-xs': '-20px',
      },
    },
  },
  safelist: [
  ],
  plugins: [
    require('tailwindcss-3d'),
    
    function ({ addUtilities }) {
      addUtilities({
      });
    }
  ],
  important: true,
};
```
</details>

If relevant, this is in a .NET 8 Blazor Interactive Server project

**Comment by theron-wang** [comment]
Thanks - I was able to reproduce the error. It turns out that the themes of the plugins were overriding the theme in the configuration file, so `primary` and `dark-blue` were not being recognized as colors. This should now be fixed in 1.5.4; if the issue still persists, please let me know!

**Comment by v36u** [comment]
Yes. Looks good now. Thanks!
